R/survey-tidiers.R

```diff
@@ -1,3 +1,62 @@
+#' @templateVar class svyglm
+#' @template title_desc_tidy_lm_wrapper
+#'
+#' @param x A `svyglm` object returned from [survey::svyglm()].
+#'
+#' @export
+#' @family lm tidiers
+#' @seealso [survey::svyglm()], [stats::glm()]
+tidy.svyglm <- tidy.lm # since tidy.glm is itself a copy of tidy.lm
```

Comment:
I'd prefer to write a custom method for `svyglm` objects rather than calling `tidy.lm()`. So many objects user the `lm` tidiers now that they are becoming very hard to maintain. I'd prefer some minor code duplication to increase the maintainability / modularity of tidiers. Would you write out a custom `tidy()` method here?

Reply:
I'll happily do so, but would it be fine with you to do this in two steps: first get this PR through, and then work on modularity? If that's fine with you, I'll close this conversation, and open a separate issue about modularity.

Reply:
Okay, done: `tidy.svyglm` is now separate from `tidy.lm` (which contains code that it does not need), although it still depends on `process_lm` and `tidy.summary.lm`, which contains exactly the code that it needs.

This PR might need to be reviewed (to ensure that the `tidy` and `glance` methods of `svyglm` make entire sense). Of course, the ideal reviewer would be @tslumley :)